src/docs/learn/architecture/test.md

```diff
@@ -0,0 +1,147 @@
+# Test Infrastructure
+
+::: info
+
+This article serves as an invitation for sharing ideas to improve our test infrastructure,
+feel free to contact me on [twitter](https://x.com/boshen_c) or [discord](https://discord.gg/9uXCAwqQZW).
+
+:::
+
+In Oxc, correctness and reliability are taken extremely seriously.
+
+We spend half of our time on strengthening the test infrastructure to prevent problems from propagating to downstream tools.
+
+## Parser
+
+### Memory Safety
+
+The oxc parser uses [`bumpalo`](https://docs.rs/bumpalo/latest/bumpalo) as the memory allocator for its AST.
+None of the AST nodes have a `drop` implementation.
+Miri [is used](https://github.com/oxc-project/oxc/actions/workflows/miri.yml) to ensure that no heap-allocated data is stored on the AST nodes
+so that memory is leaked when the memory allocator is dropped.
+
+### Fuzzing
+
+To ensure that the parser does not panic when encountering random data, three fuzzers are used:
+
+1. [cargo fuzz](https://github.com/rust-fuzz/cargo-fuzz) for [sending random bytes](https://github.com/oxc-project/oxc-fuzz-parser/blob/main/fuzz/fuzz_targets/parser.rs) to the parser.
+2. [shift-fuzzer-js](https://github.com/shapesecurity/shift-fuzzer-js) by [bakkot](https://github.com/bakkot) for producing random by valid ASTs.
+3. [Automated-Fuzzer](https://github.com/qarmin/Automated-Fuzzer) by [qarmin](https://github.com/qarmin), who [actively reports](https://github.com/oxc-project/oxc/issues?q=is%3Aissue+author%3Aqarmin+) crashes.
+
+### Conformance
+
+All parser tests from [Test262](https://github.com/tc39/test262), [Babel](https://github.com/babel/babel), and [TypeScript](https://github.com/microsoft/TypeScript) are used to test JavaScript, TypeScript, and JSX syntax.
+
+For test262, all stage 4 and regular expression tests are included.
+
+All conformance results are stored in a snapshot file for tracking changes:
```

Comment:
We should mention that every single tool has its own conformance tests, not just the parser